### PR TITLE
Feat/androidx viewmodel multiplatform

### PR DESCRIPTION
**Pull request by levinzonr**
https://monstarlab.atlassian.net/jira/software/projects/AIM/boards/396?selectedIssue=AIM-78

- ViewModels are injected rather than created inside composables
- ViewModels dependencies are part of the constructor, removing Koin details from its implementation
- ViewModel multiplatform is used instead of KMM ViewModel

### CODE REVIEW

**Review by aslamhossin — APPROVED** [review]

**Review by jhasan95 — APPROVED** [review]
LGTM!

**Comment by aslamhossin** [comment]
please ensure that KMMViewModelSwiftUI is not used within the iOS module.

File: composeApp/src/commonMain/kotlin/di/KoinCommon.kt
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/di/KoinCommon.kt:1]: We've extracted the modules to different files.
This file will only contain `initKoin()` function.
May be it was a merge issue.

**Review by ShaQib07 — CHANGES_REQUESTED** [review]

File: composeApp/src/commonMain/kotlin/di/KoinCommon.kt
Comment (levinzonr) [reply composeApp/src/commonMain/kotlin/di/KoinCommon.kt:1]: Yeah most likely! Thanks for pointing that out

Comment (levinzonr) [reply composeApp/src/commonMain/kotlin/di/KoinCommon.kt:1]: Fixed

File: composeApp/src/commonMain/kotlin/di/KoinCommon.kt [hunk 1]
```diff
@@ -21,6 +23,7 @@ fun initKoin(appModule: Module = module { }): KoinApplication = startKoin {
         dataPersistenceModule,
         repositoryModule,
         useCaseModule,
-        platformModule
+        platformModule,
+        viewModelsModule
     )
-}
+}
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/di/KoinCommon.kt]: Missing new line

File: composeApp/src/commonMain/kotlin/di/modules/ViewModelsModule.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import org.koin.dsl.module
+import presentation.feature.home.HomeViewModel
+import presentation.feature.login.LoginViewModel
+import presentation.feature.pagination.PaginatedHomeViewModel
+
+val viewModelsModule = module {
+    factory { LoginViewModel(get()) }
+    factory { HomeViewModel(get(), get(), get(), get()) }
+    factory { PaginatedHomeViewModel(get(), get()) }
+}
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/di/modules/ViewModelsModule.kt]: Missing new line

File: composeApp/src/commonMain/kotlin/presentation/feature/home/HomeViewModelProvider.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package presentation.feature.home
+
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class HomeViewModelProvider: KoinComponent {
+    private val viewModel by inject<HomeViewModel>()
+    fun provide() = viewModel
+}
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/presentation/feature/home/HomeViewModelProvider.kt]: Missing new line

File: iosApp/Podfile.lock
Comment (ShaQib07) [inline iosApp/Podfile.lock:1]: This file should be in the gitignore

File: iosApp/iosApp.xcodeproj/xcshareddata/xcschemes/iosApp.xcscheme
Comment (ShaQib07) [inline iosApp/iosApp.xcodeproj/xcshareddata/xcschemes/iosApp.xcscheme:1]: This file should be in the gitignore

**Review by ShaQib07 — CHANGES_REQUESTED** [review]

File: composeApp/src/commonMain/kotlin/di/modules/ViewModelsModule.kt (hunk 2 above)
Comment (levinzonr) [reply composeApp/src/commonMain/kotlin/di/modules/ViewModelsModule.kt]: fxied

File: composeApp/src/commonMain/kotlin/presentation/feature/home/HomeViewModelProvider.kt (hunk 3 above)
Comment (levinzonr) [reply composeApp/src/commonMain/kotlin/presentation/feature/home/HomeViewModelProvider.kt]: fixed

File: iosApp/Podfile.lock
Comment (levinzonr) [reply iosApp/Podfile.lock:1]: it already is, probably not picked up due to a broken path, fixed

File: iosApp/iosApp.xcodeproj/xcshareddata/xcschemes/iosApp.xcscheme
Comment (levinzonr) [reply iosApp/iosApp.xcodeproj/xcshareddata/xcschemes/iosApp.xcscheme:1]: fixed

**Review by ShaQib07 — APPROVED** [review]
LGTM

**Review by ShaQib07 — COMMENTED** [review]
However, this ViewModel doesn't work properly with SwiftUI yet.
I guess it'd be better If we draft this PR for now.

**Comment by ShaQib07** [comment]
Findings:
There's no issue with the ViewModel it seems.
The issue is with the StateFlow which isn't working with swift UI.
Previously we were using KMMViewModel's Mutable StateFlow with nativeCoroutine annotation.
But the default kotlin's MutableStateFlow ain't working like before.

cc: @aslamhossin

**Comment by levinzonr** [comment]
Think it should work together with SKIA, in that repo where androidx View model was used there is also SKIA 

Let's verify once SKIA work is merged

**Comment by levinzonr** [comment]
Closing in favour of https://github.com/monstar-lab-oss/KMP-Template/pull/11

**Pull request closed without merging**